Source/Experimental/AttributeArrayIteratorExample/AttributeArray.hpp

```diff
@@ -0,0 +1,610 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <functional>
+#include <numeric>
+
+// -----------------------------------------------------------------------------
+template <typename T> class AttributeArray
+{
+  public:
+
+    using comp_dims_type = std::vector<size_t>;
+
+    using size_type = size_t;
+    using value_type = T;
+    using reference = T&;
+    using iterator_category=std::input_iterator_tag;
+    using pointer = T*;
+    using difference_type = value_type;
+
+
+    AttributeArray() = default;
+
+    AttributeArray(size_t ntuples, std::string name)
+      : m_Name(std::move(name))
+    {
+      m_Array = resizeAndExtend(ntuples);
+    }
+
+    AttributeArray(size_t ntuples, comp_dims_type cdims, const std::string& name) :
+      m_NumTuples(ntuples)
+    , m_CompDims(std::move(cdims))
+    {
+      m_NumComponents = std::accumulate(m_CompDims.begin(), m_CompDims.end(), 1, std::multiplies<T>());
+      m_InitValue = static_cast<T>(0);
+      m_Array = resizeAndExtend(m_NumTuples * m_NumComponents);
+    }
+    //
+    ~AttributeArray()
+    {
+      deallocate();
+    }
+
+    AttributeArray(const AttributeArray&) = default;           // Copy Constructor Not Implemented
+    AttributeArray(AttributeArray&&) = delete;                 // Move Constructor Not Implemented
+    AttributeArray& operator=(const AttributeArray&) = delete; // Copy Assignment Not Implemented
+    AttributeArray& operator=(AttributeArray&&) = delete;      // Move Assignment Not Implemented
+
+
+    class tuple_iterator
+    {
+      public:
+        using self_type = tuple_iterator;
+        using value_type = T;
+        using reference = T&;
+        using pointer = T*;
+        using difference_type = value_type;
+        using iterator_category = std::forward_iterator_tag;
+
+        tuple_iterator(pointer ptr, size_type numComps)
+          : ptr_(ptr), num_comps_(numComps)
+        {
+        }
+        self_type operator++()
+        {
+          ptr_ = ptr_ + num_comps_;
+          return *this;
+        } // PREFIX
+        self_type operator++(int junk)
+        {
+          self_type i = *this;
+          ptr_ = ptr_ + num_comps_;
+          return i;
+        } // POSTFIX
+        reference operator*()
+        {
+          return *ptr_;
+        }
+        pointer operator->()
+        {
+          return ptr_;
+        }
+        bool operator==(const self_type& rhs)
+        {
+          return ptr_ == rhs.ptr_;
+        }
+        bool operator!=(const self_type& rhs)
+        {
+          return ptr_ != rhs.ptr_;
+        }
+        reference comp_value(size_type comp)
+        {
+          return *(ptr_ + comp);
+        }
+
+      private:
+        pointer ptr_;
+        size_t num_comps_;
+    };
+
+    class const_tuple_iterator
+    {
+      public:
+        using self_type = tuple_iterator;
```

Comment:
I believe this should be `using self_type = const_tuple_iterator;`

Reply:
yep.